cdisc_rules_engine/services/data_services/base_data_service.py

```diff
@@ -243,38 +246,37 @@ def _handle_special_cases(
             if self._contains_topic_variable(dataset, dataset_metadata.domain, "OBJ"):
                 return FINDINGS_ABOUT
             return FINDINGS
-        if self._is_associated_persons(dataset):
+        if dataset_metadata.is_ap and not dataset_metadata.is_supp:
```

Comment:
I know this probably wasn't the original behavior, but we shouldn't have `and not dataset_metadata.is_supp` here. `_get_associated_persons_inherit_class` will need to be modified so that it recognizes supps as `RELATIONSHIP` class